tests/integration_tests/modules/test_ubuntu_drivers.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from tests.integration_tests.clouds import IntegrationCloud
+from tests.integration_tests.util import verify_clean_log
+
+USER_DATA = """\
+#cloud-config
+drivers:
+    nvidia:
+        license-accepted: true
+"""
+
+
+@pytest.mark.user_data(USER_DATA)
+@pytest.mark.adhoc  # Expensive instance type
+@pytest.mark.oci
+def test_ubuntu_drivers_installed(session_cloud: IntegrationCloud):
+    with session_cloud.launch(
+        launch_kwargs={"instance_type": "VM.GPU3.1"}
+    ) as client:
+        log = client.read_from_file("/var/log/cloud-init.log")
+        verify_clean_log(log)
+        assert 1 == log.count(
+            "Installing and activating NVIDIA drivers "
+            '("nvidia/license-accepted"=True, version=latest)'
+        )
+        cmd = "apt list --installed | grep -E 'nvidia-*-server'"
+        assert client.execute(cmd).ok
```

Comment:
This command will fail because invoking apt from a script exits non-zero saying something like `don't use apt interfacefrom scripts`.

Here's an alternative (that I put into the full diff on my final comment 
```suggestion
        result = client.execute("dpkg -l | grep nvidia")
        assert result.ok, "No nvidia packages found"
        assert re.search(
            r"ii\s+linux-modules-nvidia-\d+-server", result.stdout
        ), (
            f"Did not find specific nvidia drivers packages in:"
            f" {result.stdout}"
         )
```